pkg/flock/lock_solaris.go

```diff
@@ -0,0 +1,64 @@
+// +build solaris
+
+package flock
+
+import (
+	"os"
+	"syscall"
+)
+
+var (
+	wrlck = syscall.Flock_t{
+		Type:   syscall.F_WRLCK,
+		Whence: 0,
+		Start:  0,
+		Len:    0,
+		Pid:    0,
+	}
+	rdlck = syscall.Flock_t{
+		Type:   syscall.F_RDLCK,
+		Whence: 0,
+		Start:  0,
+		Len:    0,
+		Pid:    0,
+	}
+	unlck = syscall.Flock_t{
+		Type:   syscall.F_UNLCK,
+		Whence: 0,
+		Start:  0,
+		Len:    0,
+		Pid:    0,
+	}
+)
+
+func open(path string, flag int, perm os.FileMode) (*os.File, error) {
+	return os.OpenFile(path, flag, perm)
+}
+
+func lock(f *os.File) error {
```

Comment:
Plain old F_SETLKW wouldn't conflict with a lock on a different fd on the same name (i.e, open file description) by the same minio server process.